src/fleet/azext_fleet/custom.py

```diff
@@ -717,3 +774,217 @@ def approve_gate(cmd,  # pylint: disable=unused-argument
+
+    managed_namespace_model = cmd.get_models(
+        "FleetManagedNamespace",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleet_managed_namespaces"
+    )
+
+    managed_namespace_properties_model = cmd.get_models(
+        "ManagedNamespaceProperties",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleet_managed_namespaces"
+    )
+
+    resource_quota_model = cmd.get_models(
+        "ResourceQuota",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleet_managed_namespaces"
+    )
+
+    network_policy_model = cmd.get_models(
+        "NetworkPolicy",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleet_managed_namespaces"
+    )
+
+    fleet_client = cf_fleets(cmd.cli_ctx)
+    fleet = fleet_client.get(resource_group_name, fleet_name)
+
+    default_resource_quota = None
+    if cpu_requests or cpu_limits or memory_requests or memory_limits:
+        resource_limits = {}
+        if cpu_requests:
+            resource_limits['cpu_request'] = cpu_requests
+        if memory_requests:
+            resource_limits['memory_request'] = memory_requests
+        if cpu_limits:
+            resource_limits['cpu_limit'] = cpu_limits
+        if memory_limits:
+            resource_limits['memory_limit'] = memory_limits
+        default_resource_quota = resource_quota_model(**resource_limits)
+
+    default_network_policy = None
+    if ingress_policy or egress_policy:
+        network_policies = {}
+        if ingress_policy:
+            network_policies['ingress'] = ingress_policy
+        if egress_policy:
+            network_policies['egress'] = egress_policy
+        default_network_policy = network_policy_model(**network_policies)
+
+    managed_namespace_props = managed_namespace_properties_model(
+        labels=labels,
+        annotations=annotations,
+        default_resource_quota=default_resource_quota,
+        default_network_policy=default_network_policy
+    )
+
+    propagation_policy = None
+    if member_cluster_names:
+        placement_policy = PlacementV1PlacementPolicy(
+            placement_type=PlacementType.pick_fixed,
+            cluster_names=member_cluster_names
+        )
+        placement_spec = PlacementV1ClusterResourcePlacementSpec(
+            policy=placement_policy
+        )
+        placement_profile = PlacementProfile(
+            default_cluster_resource_placement=placement_spec
+        )
+        propagation_policy = PropagationPolicy(
+            type=PropagationType.placement,
+            placement_profile=placement_profile
+        )
+
+    managed_namespace = managed_namespace_model(
+        location=fleet.location,
+        tags=tags,
+        managed_namespace_properties=managed_namespace_props,
+        adoption_policy=adoption_policy,
+        delete_policy=delete_policy,
+        propagation_policy=propagation_policy
+    )
+
+    return client.begin_create_or_update(
```

Comment:
please wrap all our long running operations with sdk_no_wait and specify polling_interval=5 to make them more interactive?